gammapy/cube/fit.py

```diff
@@ -360,23 +357,24 @@ def stack(self, other):
         """
         if self.counts and other.counts:
             self.counts.data[~self.mask_safe] = 0
-            self.counts.coadd(other.counts, weights=other.mask_safe)
+            self.counts.stack(other.counts, weights=other.mask_safe, check=False)
```

Comment:
Why `check = False`?  Could there be cases where someone tries to stack datasets which are not cutouts?

Reply:
I removed the `check` option...